python_modules/dagster/dagster/core/execution/plan/inputs.py

```diff
@@ -466,6 +468,77 @@ def compute_version(
         return dagster_type.loader.compute_loaded_input_version(config_data)
 
 
+class FromRootInputValuePlaceholder(
+    FromRootPlaceholder,
+):
+    def __init__(self, top_level_input_name: str, input_value: Any):
+        self.top_level_input_name = top_level_input_name
+        self.input_value = input_value
+
+    def to_input_source(self, leaf_input_name: str, node_handle: NodeHandle) -> StepInputSource:
+        return FromRootInputValue(
+            input_name=self.top_level_input_name,
+            leaf_input_name=leaf_input_name,
+            input_value=self.input_value,
+            node_handle=node_handle,
+        )
+
+
+@whitelist_for_serdes
+class FromRootInputValue(
+    NamedTuple(
+        "_FromRootInputValue",
+        [
+            ("input_name", str),
+            ("leaf_input_name", str),
+            ("input_value", Any),
```

Comment:
Input values won't necessarily be serializable - will that cause a problem? If it does cause a problem, maybe could address by pulling the input value off of the PipelineDefinition?

Reply:
yup that's right - we needed to pull them off the jobdefinition